src/NuGet.Core/NuGet.Packaging/Signing/Signer.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Remove or add signature package metadata.
+    /// </summary>
+    public class Signer
+    {
+        // Temporary
+        private const string TestSignedPath = "testsigned/signed.json";
+        private readonly ISignPackage _package;
+
+        /// <summary>
+        /// Creates a signer for a specific package.
+        /// </summary>
+        /// <param name="package">Package to sign or modify.</param>
+        public Signer(ISignPackage package)
+        {
+            _package = package ?? throw new ArgumentNullException(nameof(package));
+        }
+
+        /// <summary>
+        /// Add a signature to a package.
+        /// </summary>
+        public async Task SignAsync(Signature signature, ILogger logger, CancellationToken token)
```

Comment:
Shouldn't this consider the manifest file as well? In real implementation that should be hashed and put into the signature object.
But even for temp/fake implementation, it seems important to capture that information in the signatures.

Reply:
It will create the manifest